x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -126,5 +139,27 @@ public ValueFetcher valueFetcher(SearchExecutionContext context, String format)
         public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext) {
             throw new IllegalArgumentException("[semantic_text] fields do not support sorting, scripting or aggregating");
         }
+
+        public Query semanticQuery(InferenceResults inferenceResults, SearchExecutionContext context) {
```

Comment:
I think it should be exposed as query builders first. This would avoid discrepancy in the default that we use (compare to the text_expansion query for instance).

Reply:
`semanticQuery` now returns a `QueryBuilder`